datahost-ld-openapi/test/tpximpact/datahost/ldapi/handlers/delta_test.clj

```diff
@@ -0,0 +1,37 @@
+(ns tpximpact.datahost.ldapi.handlers.delta-test
+  (:require
+   [clojure.test :refer [deftest testing is]]
+   [tablecloth.api :as tc]
+   [tpximpact.test-helpers :as th]
+   [tpximpact.datahost.uris :as uris]
+   [tpximpact.datahost.ldapi.handlers.delta :as handlers.delta]
+   [tpximpact.datahost.ldapi.util.data.delta :as data.delta]
+   [tpximpact.datahost.ldapi.util.data.validation :as data.validation]))
+
+(deftest diffing-datasets
+  ;; (th/with-system-and-clean-up {http-port :tpximpact.datahost.ldapi.jetty/http-port :as sys}
+  ;;   (is false))
+  (let [row-schema (data.validation/make-row-schema-from-json
+                    {"dh:columns" [{"@type" "dh:DimensionColumn"
+                                    "csvw:name" "attr1"
+                                    "csvw:required" true
+                                    "csvw:datatype" :string
+                                    "csvw:titles" "attr1"}
+                                   {"@type" "dh:MeasureColumn"
+                                    "csvw:name" "measure"
+                                    "csvw:datatype" :double
+                                    "csvw:titles" "measure"}]})
+        ds1 (data.validation/as-dataset
+             ^{:type :datahost.types/seq-of-maps} [{"measure" 1.0 "attr1" "a"}
+                                                   {"measure" 1.0 "attr1" "b"}
+                                                   {"measure" 4.5 "attr1" "d"}] {})
+        ds2 (data.validation/as-dataset
+             ^{:type :datahost.types/seq-of-maps} [{"measure" 1.2 "attr1" "a"}
+                                                   {"measure" 3.0 "attr1" "c"}
+                                                   {"measure" 4.5 "attr1" "d"}] {})
+        result (data.delta/delta-dataset ds1 ds2 {:row-schema row-schema
+                                                      :measure-column "measure"
+                                                      :coords-columns ["attr1"]})]
+    (is (= 4 (tc/row-count result)))
+    (is (= 3 (count (set (tc/->array result "datahost.row/id")))))))
+
```

Comment:
I don't think you're wrong in the light of the keynote presentation, but this is the part I don't get and wanted to leave for the next PR dealing with deltas..

For example, a valid dataset has only one occurrence of some coords. Which translates to only one 'change' (append | retraction | correction. and corrections = retract+append, but that doesn't matter here). And delta should be applied only against particular commit (meaning, order matters). We're always doing 1 operation per coords per commit (again, the correction case shouldn't matter here). So I don't see what the 'ID' column is for?  If $` coords_1 \in D_{base} \land coords_1 \in D_{target}  `$ and $`D_{base}, D_{target}`$ have only one occurrence of $`coords_1`$ then looking at a delta,  it's obvious what the coords refer to.

Reply:
You're right that the information is redundant but, we will likely need the id column for building URI templates for observations which are hashes of the dimensions in the table.

Also I think we should change `attr1` to be `dim1` in the test data because attributes are also part of the RDF DataCube spec and these aren't attributes, but are dimensions.

Finally if we could rename `ref` to `previous` that would be more obvious; I'm happy to keep the namespaces though! 

👍